tests/std/syntax/e2eCases/roactToReact/init.luau

```diff
@@ -0,0 +1,280 @@
+				continue
+			end
+			local key = entry.key
+			local keyRepl
+			if key.index.text == "Children" then
+				keyRepl = ReactPropsTransforms["Roact.Children"](key)
+			elseif key.index.text == "Ref" then
+				keyRepl = ReactPropsTransforms["Roact.Ref"](key)
+			end
+			if keyRepl then
+				needsReactRequire = true
+				-- couldn't replace full entry, so went with this
+				t.replacements[key :: any] = keyRepl
+				t.replacements[entry.indexerclose] = ""
+				t.replacements[entry.indexeropen] = ""
+			end
+		end
+	end)
+
+	-- convert/add requires based on all the above (if we need ReactRoblox, if we need React)
+	if needsReactRequire or needsReactRobloxRequire then
+		local roactRequire = utils.getRequireStatements(ast, "Roact").nodes[1]
+		local requireCall = roactRequire.values[1].node
+		local requiredModule = requireCall.arguments[1].node
+		if needsReactRequire and needsReactRobloxRequire then
+			-- get PackagesRoot
+			local packagesRoot = printer.printnode(requiredModule.expression)
+			t.replacements[roactRequire] =
+				`local React = require({packagesRoot}.React)\nlocal ReactRoblox = require({packagesRoot}.ReactRoblox)\n`
+			-- replace whole statement
+		else
+			-- just replace index and local var name of needed package
+			t:selectFromRoot(requiredModule :: any, queryUtils.isExprIndexName)
+				:map(function(node)
+					return node.index
+				end)
+				:replace(function(node)
+					return if needsReactRequire then "React" else "ReactRoblox"
+				end)
+
+			local roactVar = roactRequire.variables[1].node
+			t:selectFromRoot(roactVar :: any, queryUtils.isToken):replace(function(node: luau.Token)
+				return if needsReactRequire then "React" else "ReactRoblox"
+			end)
+		end
+	end
+
+	return t:transform()
+end
+
+return {
+	transform = transform,
+	cases = require("./roactToReact/cases"),
+}
+
+--[[
+
+High level notes:
+
+- upwards navigation is rly useful; in this codemod, it'd be a lot easier / more performant to query for each function call (generally), update all those that are migrated accordingly (if we have upwards nav
+  we can get the statement by walking up tree from the call node). unfortunately, bc some of the migrations require the whole call statement (i.e local var = call),
+  we cannot do this and have to query for statements for those transforms that require the whole statement context, and separately query for calls for those transforms that only need the call node
+
+- I'm not sure of a good way around this one, just a general performance concern, it's tricky to vary operations between node world / string world.
+  When operating on a node, it is not uncommon to want to re-use parts of that node in your transformed output; currently, this will require:
+  	- for string transform: parsing the node into a string, and interpolating into your returned replacement
+	- for node transform: cloning node so it's mutable, updating accordingly (even more impractical, would never do this, just pointing out you CAN)
```

Comment:
I think we should be biasing towards string transforms. Most codemods are going to be one-and-done, not multi-pass.